### PR TITLE
cleanup task state file for xen

### PR DESCRIPTION
**Pull request by giggsoff**
We rely on file with status to determine the current state of domain for
 xen. But we do not clean the file from the previous run of domain, so
 may have unexpected state which come from the previous run.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: pkg/pillar/hypervisor/xen.go [hunk 1]
```diff
@@ -517,8 +519,21 @@ func (ctx xenContext) Delete(domainName string) (result error) {
 	return nil
 }
 
+// getTaskStateFilePath returns file where xen-start puts state of domain
+// must be aligned with xen-start logic
+func (ctx xenContext) getTaskStateFilePath(domainName string) string {
+	return filepath.Join("/run/tasks", domainName)
+}
+
 // Cleanup removes containerd-shim
```
Comment (eriknordmark) [inline pkg/pillar/hypervisor/xen.go:528]: This comment isn't correct with your changes.

**Review by eriknordmark — COMMENTED** [review]
Have you seen failures which might be caused by reading a stale file?

File: pkg/pillar/hypervisor/xen.go (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/hypervisor/xen.go]: Correct, fixed, thank you

**Comment by giggsoff** [comment]
> Have you seen failures which might be caused by reading a stale file?

I am trying to fix permanent ROL failures for xen, for example https://github.com/lf-edge/eve/actions/runs/3566856698/jobs/5993823876#step:18:5083. The stale file looks like a possible reason for me, but we will run tests with @Insei in fork to double check is it the solution.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**